src/components/pagination/pagination.ts

```diff
@@ -0,0 +1,301 @@
+import {
+  Component,
+  Input,
+  Output,
+  EventEmitter,
+  ElementRef,
+  DynamicComponentLoader,
+  AfterViewInit,
+  AfterContentInit
+} from '@angular/core';
+import {isPresent} from '@angular/core/src/facade/lang';
+import 'rxjs/add/operator/filter';
+import {PaginationService} from './pagination_service';
+
+export interface IPaginationModel {
+  currentPage: number;
+  itemsPerPage: number;
+  totalItems: number;
+}
+
+export abstract class AbstractPaginationSubComponent {
+
+  @Input()
+  name: string;
+
+  @Input()
+  model: IPaginationModel = {
+    currentPage: 0,
+    itemsPerPage: 0,
+    totalItems: 0
+  };
+
+  constructor(protected service: PaginationService) {
+    this.service.onChange
+      .filter(event => isPresent(event) && isPresent(event.name))
+      .filter(event => event.target === this.name)
+      .subscribe(event => {
+        this.model = event.pagination;
+      });
+  }
+
+}
+
+@Component({
+  selector: 'md-pagination-range',
+  template: '',
+  host: {
```

Comment:
ng2 styleguide now calling for using `@HostBinding()` instead of the `host` property in Style 06-03